src/charm.py

```diff
@@ -772,6 +743,22 @@ def _prometheus_version(self) -> Optional[str]:
             return result
         return result.group(1)
 
+    def pull(self, path) -> Optional[str]:
+        """Pull file from container (without raising pebble errors).
+
+        Returns:
+            File contents if exists; None otherwise.
+        """
+        try:
+            return self.container.pull(path, encoding="utf-8").read()
+        except (FileNotFoundError, PebbleError):
+            # Drop FileNotFoundError https://github.com/canonical/operator/issues/896
+            return None
+
+    def push(self, path, contents):
+        """Push file to container, creating subdirs as necessary."""
+        self.container.push(path, contents, make_dirs=True, encoding="utf-8")
+
```

Comment:
These two methods seems private, I would name them: `_push` and `_pull`

Reply:
Done.